src/list.alp

```diff
@@ -0,0 +1,307 @@
+{-| A library for manipulating lists of values.
+-}
+
+module list
+
+export head/1, tail/1, isEmpty/1, map/2, any/2, all/2, reverse/1, length/1,
+       concat/1, filter/2, append/2, foldLeft/3, foldRight/3, sort/1
+
+import_type option.option
+
+
+{-| Extract the first element of a list
+
+      head [3, 4] == (Some 3)
+      head [] == (None)
+-}
+val head 'a : fn (list 'a) -> option 'a
```

Comment:
I have a rough preference for `head` to explode with a bad match exception on empty lists while something like `headOption/1` would return `option 'a` but also admit that's simply what I'm used to.  Thoughts?

Reply:
I would prefer the most obvious functions to be total and have `unsafeHead` for the version that throws.

Reply:
OK, that makes sense to me.